src/jit/gentree.h

```diff
@@ -1747,6 +1753,15 @@ struct GenTree
     // sets "*pIsEntire" to true if this assignment writes the full width of the local.
     bool DefinesLocalAddr(Compiler* comp, unsigned width, GenTreeLclVarCommon** pLclVarTree, bool* pIsEntire);
 
+    regNumber RegAtDef() const
+    {
+        return GetRawRegNum();
+    }
+    regNumber RegAtUse() const
+    {
+        return isContained() ? REG_NA : GetRawRegNum();
+    }
+
```

Comment:
I agree with Siva's points.
In any case, I don't think this kind of logic belongs in `GenTree` because it is highly phase-dependent. Introducing this into `GenTree` in my opinion exacerbates existing problems with "containedness" and its phase-dependence.

---
In reply to: [85572738](https://github.com/dotnet/coreclr/pull/7876#discussion_r85572738) [](ancestors = 85572738)